search2/python/search/save/handler.py

```diff
@@ -212,21 +196,21 @@ def _process_query_chunk(query_args: dict, search_client_args: dict, field_names
 
     return resp["hits"]["total"]["value"]
 
-
-# TODO: get max_slices from opensearch index settings
-def go(
+async def go( # pylint:disable=invalid-name
```

Comment:
This method is also over 100 lines long, and it would be good to break up into smaller meaningful chunks.  Part of what makes it challenging is that we're operating at several different levels of abstraction at once, ranging from manipulating queries to building file paths to configuring file compression parameters.  Keeping this method at the highest common level of abstraction and suppressing the incidental details, farming out to helper methods as necessary, would go a long way towards readability imho.

Reply:
We've discussed this,